pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/AggregateOperator.java

```diff
@@ -159,12 +155,16 @@ public String toExplainString() {
   @Override
   protected TransferableBlock getNextBlock() {
     try {
-      if (!_readyToConstruct && !consumeInputBlocks()) {
-        return TransferableBlockUtils.getNoOpTransferableBlock();
+      TransferableBlock finalBlock;
+      if (_isGroupByAggregation) {
+        finalBlock = consumeGroupBy();
```

Comment:
This is a lateral change, not completely necessary, but helps readibility, removes a condition inside the loom and makes it easier the JIT compilation